### PR TITLE
vim-patch:8.1.{471,723,1235,1490,1951,2183,2269,2329,2330}

### PR DESCRIPTION
**Pull request by janlazo**
@blueyed DId I miss any vim patch?

### CODE REVIEW

File: src/nvim/tag.c [hunk 1]
```diff
@@ -1254,15 +1252,7 @@ static int find_tagfunc_tags(
       break;
     }
 
-    if (name_only) {
-      mfp = vim_strsave(res_name);
-    } else {
-      mfp = (char_u *)xmalloc((int)sizeof(char_u) + len + 1);
-    }
-
-    if (mfp == NULL) {
```
Comment (janlazo) [inline src/nvim/tag.c]: This shouldn't happen but the clang and pvs reports don't mention it.

File: src/nvim/tag.c [hunk 2]
```diff
@@ -1254,15 +1252,7 @@ static int find_tagfunc_tags(
       break;
     }
 
-    if (name_only) {
-      mfp = vim_strsave(res_name);
-    } else {
-      mfp = (char_u *)xmalloc((int)sizeof(char_u) + len + 1);
```
Comment (janlazo) [inline src/nvim/tag.c]: When is `sizeof(char_u)` not equal to 1 in C99?

Comment (justinmk) [reply src/nvim/tag.c]: regardless, sizeof() in malloc() is just good practice to show the intent (though in Nvim codebase we would use `sizeof(*foo)`

**Comment by justinmk** [comment]
> Developer can omit '.res' suffix now.
> TEST_FILE=test_syntax make oldtest

Nice. Will mention that in the "doc" PR in test/README.md.

**Comment by blueyed** [comment]
> @blueyed DId I miss any vim patch?

Sorry for not replying earlier.
JFI: 8.1.0723 (and 8.1.1490) was included in https://github.com/neovim/neovim/pull/10650 already, but it's hanging there since for too long already - not your fault.. ;)
I've amended it there, and will work on finishing it finally.

> > Developer can omit '.res' suffix now.
> > TEST_FILE=test_syntax make oldtest
> 
> Nice. Will mention that in the "doc" PR in test/README.md.

In https://github.com/neovim/neovim/pull/10650#discussion_r316548651 you've preferred `TEST_FILE=test_syntax.vim`.  With #10650 all variants should be supported, but I think docs should prefer the one with the suffix (actual filename).

**Comment by justinmk** [comment]
> you've preferred `TEST_FILE=test_syntax.vim`. With #10650 all variants should be supported, but I think docs should prefer the one with the suffix (actual filename).

yes, i still prefer that but it's not clear to me if that will be more or less burden. And it doesn't work currently...

**Pull request merged**